sys/include/suit.h

```diff
@@ -0,0 +1,225 @@
+ * @param[out]  buf         buffer to write into
+ * @param       len         size of the buffer
+ *
+ * @return                  Length of the url
+ * @return                  negative on error
+ */
+ssize_t suit_get_url(const suit_manifest_t *manifest, char *buf, size_t len);
+
+/**
+ * @brief Copy the version from the manifest into the supplied uint32_t
+ *
+ * @param       manifest    manifest to retrieve the version from
+ * @param[out]  version     version number to set
+ *
+ * @return                  @ref SUIT_OK on success
+ * @return                  negative on error
+ */
+int suit_get_version(const suit_manifest_t *manifest, uint32_t *version);
+
+/**
+ * @brief Retrieve the sequence number of the manifest
+ *
+ * @param       manifest    manifest to retrieve the sequence number from
+ * @param[out]  seq_no      sequence number
+ *
+ * @return                  @ref SUIT_OK on success
+ * @return                  negative on error
+ */
+int suit_get_seq_no(const suit_manifest_t *manifest, uint32_t *seq_no);
+
+/**
+ * @brief Retrieve the size from the payloadinfo in the manifest
+ *
+ * @param       manifest    manifest to retrieve the size from
+ * @param[out]  size        Size of the firmware
+ *
+ * @return                  @ref SUIT_OK on success
+ * @return                  negative on error
+ */
+int suit_payload_get_size(const suit_manifest_t *manifest, uint32_t *size);
+
+/**
+ * @brief Retrieve the digest algorithm from the payloadinfo in the manifest
+ *
+ * @param       manifest    manifest to retrieve the algo from
+ * @param[out]  algo        algorithm used
+ *
+ * @return                  @ref SUIT_OK on success
+ * @return                  negative on error
+ */
+int suit_payload_get_digestalgo(const suit_manifest_t *manifest,
+                                suit_digest_t *algo);
+
+/**
+ * @brief Retrieve the storage id from the payloadinfo in the manifest
+ *
+ * @param           manifest    manifest to retrieve the storage id from
+ * @param[inout]    buf         buffer to write the storage id in
+ * @param[inout]    len         size of the buffer and length of the storage id
+ *
+ * @return                      @ref SUIT_OK on success
+ * @return                      negative on error
+ */
+int suit_payload_get_storid(const suit_manifest_t *manifest, uint8_t *buf, size_t *len);
+
+/**
+ * @brief Retrieve the digest from the payloadinfo in the manifest
+ *
+ * @param           manifest    manifest to retrieve the digest from
+ * @param           digest      type of digest to retrieve
+ * @param[inout]    buf         buffer to write the digest in
+ * @param[inout]    len         size of the buffer and length of the digest
+ *
+ * @return                      @ref SUIT_OK on success
+ * @return                      negative on error
+ */
+int suit_payload_get_digest(const suit_manifest_t *manifest,
+            suit_digest_type_t digest, uint8_t *buf, size_t *len);
+
+#ifdef __cplusplus
+}
+#endif
+/** @} */
```

Comment:
maybe move this line below the next - so it is

```
/** @{ */
#header guard open
...
#header guard close
/* @} */
```

(if you have the file open anyways)

Reply:
Done